src/MongoDB/Cursor.c

```diff
@@ -96,7 +96,15 @@ static void php_phongo_cursor_iterator_move_forward(zend_object_iterator *iter T
 	const bson_t               *doc;
 
 	php_phongo_cursor_free_current(cursor);
-	cursor->current++;
+
+	/* If the cursor has already advanced, increment its position. Otherwise,
+	 * the first call to mongoc_cursor_next() will be made below and we should
+	 * leave its position at zero. */
+	if (cursor->advanced) {
+		cursor->current++;
+	} else {
+		cursor->advanced = true;
+	}
```

Comment:
Future improvement: refactor `phongo_cursor_advance_and_check_for_error()` so that it can take an optional `bson_t *`. Then `phongo_execute_query()` and `php_phongo_cursor_iterator_rewind()` could pass in `null`, since they only care about error checking and don't need to access the document. Meanwhile, `php_phongo_cursor_iterator_move_forward()` could pass in a `bson_t *` and, on success, use that to populate the current element, as is done with:

```c
php_phongo_bson_to_zval_ex(bson_get_data(doc), doc->len, &cursor->visitor_data);
```